task/upload.go

```diff
@@ -35,80 +35,126 @@ var (
 	OutputNameIPFSSourceMP4 = OutputName("ipfs_source_mp4")
 )
 
-func TaskUpload(tctx *TaskContext) (*TaskHandlerOutput, error) {
+type handleUploadVODParams struct {
+	tctx                     *TaskContext
+	inUrl                    string
+	getOutputLocations       func() ([]clients.OutputLocation, error)
+	finalize                 func(callback *clients.CatalystCallback) (*TaskHandlerOutput, error)
+	catalystPipelineStrategy pipeline.Strategy
+}
+
+func handleUploadVOD(p handleUploadVODParams) (*TaskHandlerOutput, error) {
 	var (
-		ctx    = tctx.Context
-		step   = tctx.Step
-		params = *tctx.Task.Params.Upload
+		ctx  = p.tctx.Context
```

Comment:
**nit:** Could have a local `tctx` variable here as well to avoid all the `p.tctx` changes below. Not a necessary change.

Reply:
Updated